addons/base_address_city/models/res_partner.py

```diff
@@ -39,7 +39,7 @@ def _fields_view_get_address(self, arch):
                     attrs="{
                         'invisible': [('country_enforce_cities', '=', True), '|', ('city_id', '!=', False), ('city', 'in', ['', False ])],
                         'readonly': [('type', '=', 'contact')%(parent_condition)s]
-                    }"
+                    }"%(required)s
```

Comment:
Works for `city_id` too thanks to the onchange. The field doesn't appear as required, but you can't validate it thanks to the onchange, and the toast message is clear enough.